### PR TITLE
build: Use vcpkg

### LINKED ISSUE #3866: Add support for CMake toolchain file in update_deps.py to ease cross compilation
Hi,

I need to cross compile the Vulkan validation layers for a ARM64 Linux target system. Therefore, I need to set the  -DCMAKE_TOOLCHAIN_FILE command line option when invoking cmake.
The script **update_deps.py** doesn't support to set this cmake command line option.
Can you please add support for this.

You can find my modified **update_deps.py** in the linked pull request.

Regards,
Oliver

### LINKED ISSUE #4614: Warnings from macOS build "built for newer macOS version (11.7) than being linked (10.12)"
MacOs builds on CI are seeing multiple linker warnings. Here is an example:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/actions/runs/3160572537/jobs/5145179242
```
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/glslang/build/install/lib/libSPVRemapper.a(SPVRemapper.cpp.o)) was built for newer macOS version (11.7) than being linked (10.12)
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/glslang/build/install/lib/libSPIRV.a(GlslangToSpv.cpp.o)) was built for newer macOS version (11.7) than being linked (10.12)
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/glslang/build/install/lib/libMachineIndependent.a(ShaderLang.cpp.o)) was built for newer macOS version (11.7) than being linked (10.12)
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/googletest/build/install/lib/libgtest.a(gtest-all.cc.o)) was built for newer macOS version (11.7) than being linked (10.12)
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/SPIRV-Tools/build/install/lib/libSPIRV-Tools.a(binary.cpp.o)) was built for newer macOS version (11.7) than being linked (10.12)
ld: warning: object file (/Users/runner/work/Vulkan-ValidationLayers/Vulkan-ValidationLayers/external/SPIRV-Tools/build/install/lib/libSPIRV-Tools.a(table.cpp.o)) was built for newer macOS version (11.7) than being linked (10.12)
```

**Environment:**
 - OS:  MacOS

This is occurring due to `CMAKE_OSX_DEPLOYMENT_TARGET` not being properly propagated. Unsure of severity of warning.

### PR DESCRIPTION
**Pull request by juan-lunarg**
Using vcpkg would replace update_deps.py

- Use industry standard
- Works much better with CMake
- Very fast
- Supports multi-configuration generators
- Cleaner builds (vcpkg detects errors very quickly)
- Custom ports in case upstream doesn't work

closes #4614
closes #3866

### CODE REVIEW

**Comment by juan-lunarg** [comment]
Using `vcpkg` instead of `update_deps.py` closes the aforementioned issues because vcpkg can properly propagate `toolchain` files to dependencies, and properly propagate other CMake state information that is important to dependencies like `CMAKE_OSX_DEPLOYMENT_TARGET`

vcpkg works on all platforms (as long as the dependency uses CMake) since it builds it from scratch.

**Comment by juan-lunarg** [comment]
note: While I raise the cmake min that's for FetchContent and nothing else. Technically I could write around that.

**Comment by juan-lunarg** [comment]
vcpkg also has fantastic binary caching. Save a lot of times on builds.

```
Restored 8 package(s) from /home/juan/.cache/vcpkg/archives in 4.336 s. Use --debug to see more details.
```

File: cmake/ports/glslang/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+vcpkg_cmake_configure(
+  SOURCE_PATH "${SOURCE_PATH}"
+  OPTIONS
+    -DSKIP_GLSLANG_INSTALL=OFF
+    -DBUILD_EXTERNAL=OFF
+    -DENABLE_GLSLANG_BINARIES=${BUILD_BINARIES}
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake)
```
Comment (lunarpapillo) [inline cmake/ports/glslang/portfile.cmake:31]: I don't really belong here, but I'm curious... what does `vcpkg_cmake_config_fixup()` do, and why is it needed?

Comment (juan-lunarg) [reply cmake/ports/glslang/portfile.cmake:31]: I mainly copied the existing port on this one. But essentially vcpkg has a standard way of laying out it's installs in the cmake binary directory. And from what I can tell the glslang puts things in a slightly different manner.

**Comment by juan-lunarg** [comment]
Closing this draft PR. I'll still leave the branch for reference

**Pull request closed without merging**